File: OptiCore.Domain/Commissions/Commission.cs
using OptiCore.Domain.Companies;
using OptiCore.Domain.Core;
using OptiCore.Domain.Users;

namespace OptiCore.Domain.Commissions
{
    public class Commission : AuditEntity
    {
        public int CompanyId { get; set; }
        public decimal Amount { get; set; }
        public int Level { get; set; }

        // Navigation property for EF Core
        public Company Company { get; set; }
    }
}

File: OptiCore.UI/src/app/models/commission.ts
import Company from './company';

interface Commission {
  id: number;
  createdOn: string;
  createdBy: string;
  tenantId: number;
  updatedOn: string;
  updatedBy: string;
  companyId: number;
  amount: number;
  level: number;
  company: Company;
}

export default Commission;
